python/ray/train/mosaic/_mosaic_utils.py

```diff
@@ -0,0 +1,167 @@
+from __future__ import annotations
+import math
+from typing import Any, Dict, Tuple, Union, Iterable
+
+from ray.air.checkpoint import Checkpoint
+from ray.air import session
+from ray.data.dataset import Dataset
+
+from composer.loggers import Logger
+from composer.loggers.logger import LogLevel
+from composer.loggers.logger_destination import LoggerDestination
+from composer.core.state import State
+from composer.callbacks.checkpoint_saver import CheckpointSaver
+
+
+class _mosaic_iterator:
+    def __init__(self, dataset, batch_size, labels):
+        self.dataset = dataset
+        self.labels = labels
+        self.total_samples = dataset.count()
+        self.batch_iter = self.dataset.iter_torch_batches(batch_size=batch_size)
+
+    def __next__(self):
+        next_data = next(self.batch_iter)
+        return [next_data[label] for label in self.labels]
+
+
+class RayDatasetMosaicIterable:
+    def __init__(self, dataset, batch_size, labels):
+        self.dataset = dataset
+        self.batch_size = batch_size
+        self.labels = labels
+        self.total_samples = dataset.count()
+
+    def __len__(self):
+        return math.ceil(self.dataset.count() / self.batch_size)
```

Comment:
Use `total_samples` instead of calling count again?

Reply:
fixed!